skimage/transform/_warps.py

```diff
@@ -898,3 +898,149 @@ def coord_map(*args):
     _clip_warp_output(image, warped, order, mode, cval, clip)
 
     return warped
+
+
+def _linear_polar_mapping(output_coords, k_ang, k_rad, center):
+    """Inverse mapping function to convert from cartesion to polar coordinates
+
+    Parameters
+    ----------
+    output_coords : ndarray
+        `(M, 2)` array of `(col, row)` coordinates in the output image
+    k_ang : float
+        Scaling factor that relates the intended number of rows in the output
+        image to angle: ``k_ang = nrows / (2 * np.pi)``
+    k_rad : float
+        Scaling factor that relates the radius of the circle bounding the
+        area to be transformed to the intended number of columns in the output
+        image: ``k_rad = ncols / radius``
+    center : tuple (row, col)
+        Coordinates that represent the center of the circle that bounds the
+        area to be transformed in an input image.
+
+    Returns
+    -------
+    coords : ndarray
+        `(M, 2)` array of `(col, row)` coordinates in the input image that
+        correspond to the `output_coords` given as input.
+    """
+    angle = output_coords[:, 1] / k_ang
+    rr = ((output_coords[:, 0] / k_rad) * np.sin(angle)) + center[0]
+    cc = ((output_coords[:, 0] / k_rad) * np.cos(angle)) + center[1]
+    coords = np.column_stack((cc, rr))
+    return coords
+
+
+def _log_polar_mapping(output_coords, k_ang, k_rad, center):
+    """Inverse mapping function to convert from cartesion to polar coordinates
+
+    Parameters
+    ----------
+    output_coords : ndarray
+        `(M, 2)` array of `(col, row)` coordinates in the output image
+    k_ang : float
+        Scaling factor that relates the intended number of rows in the output
+        image to angle: ``k_ang = nrows / (2 * np.pi)``
+    k_rad : float
+        Scaling factor that relates the radius of the circle bounding the
+        area to be transformed to the intended number of columns in the output
+        image: ``k_rad = width / np.log(radius)``
+    center : tuple (row, col)
+        Coordinates that represent the center of the circle that bounds the
+        area to be transformed in an input image.
+
+    Returns
+    -------
+    coords : ndarray
+        `(M, 2)` array of `(col, row)` coordinates in the input image that
+        correspond to the `output_coords` given as input.
+    """
+    angle = output_coords[:, 1] / k_ang
+    rr = ((np.exp(output_coords[:, 0] / k_rad)) * np.sin(angle)) + center[0]
+    cc = ((np.exp(output_coords[:, 0] / k_rad)) * np.cos(angle)) + center[1]
+    coords = np.column_stack((cc, rr))
+    return coords
+
+
+def warp_polar(image, center=None, radius=None, output_shape=None,
+               scaling='linear', **kwargs):
+    """Remaps image to polor or log-polar coordinates space.
+
+    Parameters
+    ----------
+    image : ndarray (2-D or 3-D)
```

Comment:
Is this API compatible with the desire to have things thinking about ND images when they get merged in?

It seems like we often like the "multichannel" parameter to be passed in when the last index is image channel.

Otherwise, we apply the transformation to the last two dimensions.

Not sure what is best. @jni?

Reply:
Oh yeah, 100%, we don't do this kind of guessing anymore. For multichannel images, we should require `multichannel=True` to be given.

Reply:
To be a little more explicit, jni means that by default, multichannel should be set to False. In your code, by default, images that include RGB channels should be rejected.

They should be accepted only when the multichannel argument is true